### PR TITLE
Properly support defining env vars from secret/configmap keys

### LINKED ISSUE #9629: Kubernetes env vars generation changed unexpectedly
**Describe the bug**
I have these lines in my `application.properties`:

```properties
quarkus.openshift.env-vars.DB_DATABASE.secret=postgresql
quarkus.openshift.env-vars.DB_DATABASE.value=database
```

And I expect Quarkus Kubernetes to generate an env var that corresponds to the `database` field of the `postgresql` secret. This is how it works in Quarkus 1.3.3, at least:

```yaml
        - name: "DB_DATABASE"
          valueFrom:
            secretKeyRef:
              key: "database"
              name: "postgresql"
              optional: false
```

However, when trying on Quarkus 1.5.0.CR1, I only get this:

```yaml
        - name: "DB_DATABASE"
          value: "database"
```

**Expected behavior**
See above.

**Actual behavior**
See above.

**To Reproduce**
1. Add `quarkus-openshift`.
2. Add configuration per above to `application.properties`.
3. `mvn clean package`
4. See `target/kubernetes/openshift.yml`

**Configuration**
See above.

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`:
    ```
    lthon@argondie:~$ uname -a
    Linux argondie 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
    ```
 - Output of `java -version`: 
    ```
    lthon@argondie:~$ java -version
    openjdk version "1.8.0_252"
    OpenJDK Runtime Environment (build 1.8.0_252-8u252-b09-1~18.04-b09)
    OpenJDK 64-Bit Server VM (build 25.252-b09, mixed mode)
    ```
 - GraalVM version (if different from Java): N/A
 - Quarkus version or git rev: 1.5.0.CR1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`):
    ```
    lthon@argondie:~$ mvn --version
    Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
    Maven home: /home/lthon/software/apache-maven
    Java version: 1.8.0_252, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
    Default locale: cs_CZ, platform encoding: UTF-8
    OS name: "linux", version: "4.15.0-101-generic", arch: "amd64", family: "unix"
    ```

**Additional context**
See also https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Kubernetes.20env.20vars

### PR DESCRIPTION
**Pull request by metacosm**
Fixes #9629 

- [x] Restore (undocumented) behavior that was broken with the new style configuration
- [x] Provide new style configuration for defining env vars from secret/configmap keys
- [x] Check/improve validation
- [x] Add integration tests 
- [x] Update documentation

### CODE REVIEW

**Comment by metacosm** [comment]
@iocanel this is ready for review

**Review by iocanel — CHANGES_REQUESTED** [review]
Looks great.

The only thing I'd like to be added is to slightly extend the coverage. See details in the comments.

File: integration-tests/kubernetes/quarkus-standard-way/src/test/resources/kubernetes-with-env-from-configmap.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+quarkus.kubernetes.env.configmaps=my-configmap
```
[the remark this answers is not in the document]
Comment (metacosm) [reply integration-tests/kubernetes/quarkus-standard-way/src/test/resources/kubernetes-with-env-from-configmap.properties:1]: I assume you're talking about the `1.3` branch, right? Commit ccb3574d17034ee6b09098eb5b51cdeac2829d90?

**Comment by geoand** [comment]
@metacosm is this ready? If so, we should try and get it in tomorrow's release.

@iocanel can you please review?

**Comment by metacosm** [comment]
> @metacosm is this ready? If so, we should try and get it in tomorrow's release.
> 
> @iocanel can you please review?

That's ready from my side. I just need to squash the last commits. One thing I'm not too sure about are the `vars-from-configmap` and `vars-from-secret` names which kind of are a mouthful so I'm willing to change them if people have better names for this… :)

**Comment by geoand** [comment]
I absolutely suck at naming, so I won't be able to provide any better one :)

**Comment by geoand** [comment]
@metacosm can you please squash the commits in order  to make the commit ready for merging?

**Comment by metacosm** [comment]
@geoand done

**Comment by geoand** [comment]
Thanks!

**Comment by metacosm** [comment]
Decision was made to use
```properties
quarkus.kubernetes.env.mapping.db-password.from-secret=db-secret
quarkus.kubernetes.env.mapping.db-password.with-key=database.password
```
syntax.

**Comment by geoand** [comment]
@iocanel please review since you've been following more than me

**Review by iocanel — APPROVED** [review]
LGTM

**Comment by metacosm** [comment]
I guess I didn't go through the Java 14 tests… 🤷

**Comment by geoand** [comment]
Java 14 has been a thorn in our side the past few days...

**Comment by metacosm** [comment]
> Java 14 has been a thorn in our side the past few days when it comes to CI...

Yep, I know 😉

**Pull request merged**